packages/nimble-blazor/NimbleBlazor/Components/NimbleTable.razor.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace NimbleBlazor;
+
+/// <summary>
+/// A table component
+/// </summary>
+/// <typeparam name="TData">Represents the type for a row of data in the table (an element of the <see cref="Data"/>).</typeparam>
+/// <remarks>The type represented by <see cref="TData"/> should not have any hierarchy. All aspects that can be serialized
+/// should be at the top-level.</remarks>
+public partial class NimbleTable<TData> : ComponentBase
+{
+    private ElementReference _table;
+    private bool _shouldRender = false;
+    private IEnumerable<TData> _data = Enumerable.Empty<TData>();
+    internal static string SetTableDataMethodName = "NimbleBlazor.Table.setData";
+
+    [Inject]
+    private IJSRuntime? JSRuntime { get; set; }
+
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the data for the table.
+    /// </summary>
+    [Parameter]
+    public IEnumerable<TData> Data
+    {
+        get
+        {
+            return _data;
+        }
+        set
+        {
+            _data = value;
+            _shouldRender = true;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a callback that's invoked when the data changes
+    /// </summary>
+    [Parameter]
+    public EventCallback<IEnumerable<TData>> DataChanged { get; set; }
```

Comment:
It doesn't look like anything calls this right now - is the intent that the Data property setter does it? (That would be different from most of the controls where the value change is triggered by a user interaction on the web page though)

Reply:
(I'd probably just remove it for now)

Reply:
As I mentioned to Milan this is required for data binding.